### DISCUSSION TITLE
internalSubscriptions and autobatch

### DISCUSSION THREAD

**Discussion by denbezrukov** [reduxjs/redux-toolkit#3341] [Q&A] [answered]
Hi Team.
Thank you for your work.

I was wondering if it might be a problem if `subscriptionsUpdated` make [autobatch](https://redux-toolkit.js.org/api/autoBatchEnhancer)

https://github.com/reduxjs/redux-toolkit/blob/23a78cabae33ffc514602e9ea5763446db02e1b6/packages/toolkit/src/query/core/buildSlice.ts#L426-L434

```
export const autobatchMiddleware =
  () =>
  () =>
  (next: (action: PayloadAction<unknown>) => PayloadAction<unknown>) =>
  (action: PayloadAction<unknown, string, {[SHOULD_AUTOBATCH]?: boolean}>) => {
    if (action.type.includes('internalSubscriptions/subscriptionsUpdated')) {
      action.meta ??= {}
      action.meta[SHOULD_AUTOBATCH] = true
    }

    return next(action)
  }
```

**Comment by phryneas** [comment, marked as answer]
Generally that should not cause any problems. This slice is only there for the Devtools to pick it up; it doesn't have a meaning for any runtime code. Why do you think so?

**Reply by phryneas** [reply]
Also, even when autobatching, store updates happen immediately, only the notificiation for an UI update is batched.

**Reply by denbezrukov** [reply]
Thank you for your answer!
I've just discovered that my application has a performance issue with this action and marking it autobatch solves the problem.
I was concerned that it could break the internal code.

**Reply by phryneas** [reply]
@markerikson did we really miss marking *that* action? 😅

**Reply by phryneas** [reply]
Makes sense! It's completely harmless to mark that one :)
